server/game/core/ongoingEffect/OngoingEffect.ts

```diff
@@ -29,7 +34,24 @@ const EnumHelpers = require('../utils/EnumHelpers');
  *                        and the numerical value of the effect, if any.
  */
 class OngoingEffect {
-    constructor(game, source, properties, effectImpl) {
+    public game: Game;
+    public source: Card;
+    // TODO: Can we make GameObject more specific? Can we add generics to the class for AbilityContext?
+    public matchTarget: GameObject | ((target: GameObject, context: AbilityContext) => boolean);
```

Comment:
I think the most specific we can (and should) do is `Player | Card`, and then we could specify just one or the other in an override in the child classes